lib/pos_mempool.go

```diff
@@ -663,7 +840,9 @@ func (mp *PosMempool) GetAugmentedUniversalView() (*UtxoView, error) {
 	if !mp.IsRunning() {
 		return nil, errors.Wrapf(MempoolErrorNotRunning, "PosMempool.GetAugmentedUniversalView: ")
 	}
-	newView, err := mp.readOnlyLatestBlockView.CopyUtxoView()
+	mp.augmentedLatestBlockViewMutex.RLock()
```

Comment:
You don't need to hold the lock when you're making a copy. Rewrite this as follows:
```
var readOnlyPointer *UtxoView
mp.augmentedLatestBlockViewMutex.RLock()
readOnlyPointer = mp.augmentedLatestBlockViewMutex
mp.augmentedLatestBlockViewMutex.RUnlock()

newView, err := readOnlyPointer.CopyUtxoView()
```

Note that we can do this because, unlike with the legacy mempool, we are *never modifying* augmentedLatestBlockView ever. This seems like a small change, but I think it will dramatically increase efficiency because it means we're almost never holding the lock. This change is also the reason why I recommend we decrease the wait time between recomputations of the augmented view.

Reply:
Update: Resolved this myself.